test/cmocka/src/notifier_mocks.c

```diff
@@ -0,0 +1,102 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2019 Intel Corporation. All rights reserved.
+//
+// Author: Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/notifier.h>
+#include <sof/list.h>
+#include <sof/common.h>
+
+struct callback_handle {
+	void *receiver;
+	void *caller;
+	void (*cb)(void *arg, enum notify_id, void *data);
+	struct list_item list;
+};
+
+static struct notify *_notify;
+
+struct notify **arch_notify_get(void)
+{
+	int i;
+
+	if (!_notify) {
+		_notify = malloc(sizeof(*_notify));
+
+		for (i = 0; i < NOTIFIER_ID_MAX; i++)
+			list_init(&_notify->list[i]);
+	}
+
+	return &_notify;
+}
+
+void notifier_event(void *caller, enum notify_id type, uint32_t core_mask,
+		    void *data, uint32_t data_size)
+{
+	struct notify *notify = *arch_notify_get();
+	struct list_item *wlist;
+	struct list_item *tlist;
+	struct callback_handle *handle;
+
+	list_for_item_safe(wlist, tlist, &notify->list[type]) {
```

Comment:
no need for "safe"

Reply:
It's a mock, nobody really cares, and this also reflects how original implementation looks like.